setup/README.md

```diff
@@ -56,81 +56,76 @@ Before you can finally set up Connaisseur, you will need to configure its connec
-
-##### manual
+For now let's assume you only want to verify image signatures you created/will create, using the key generated in step [two](#set-up-docker-content-trust). To get the public part of this you have to convert the key manually.
 
 - Go to your private docker trust folder `~/.docker/trust/private` and identify the root key. To do so, you can either go through the files and search for the one with `role: root` or `grep -r 'role: root'`.
-- Copy this file to a new file `root-priv.key` and remove line with `role: root` and the following empty line.
+- Copy the root key file to a new file `root-priv.key` and remove line with `role: root` and the following empty line.
 - Generate the corresponding public key via `openssl ec -in root-priv.key -pubout -out root.pub`. You will be asked to enter your root password.
-- The new `root.pub` contains your public which you copy and set for `notary.rootPubKey` in `connaisseur/helm/values.yaml`. The result should look similar as for the first case above.
+- The new `root.pub` contains your public key which you copy and set as a new `rootKeys` entry in your notary configuration of choice in `connaisseur/helm/values.yaml`. The result should look similar as in the example above.
 - To clean up, remove the `root-priv.key` and `root.pub` in `~/.docker/trust/private`.
 
-##### bash/sh/zsh
-
-- Run the script below from the Connaisseur repository to generate your root public key. You will be asked to enter your root password set above:
+Getting the public key can also be summarized in a small script:
 
 ```bash
 cd ~/.docker/trust/private
 sed '/^role:\sroot$/d' $(grep -iRl "role: root$" .) > root-priv.key
 openssl ec -in root-priv.key -pubout -out root.pub
 ```
 
-- After entering your password, copy the public key to the `helm/values.yaml`: 
+#### Configure the image policy
 
-```bash
-yq e '.notary.rootPubKey=$(cat root.pub)' -i "${OLDPWD}/helm/values.yaml"
-rm root-priv.key root.pub
-cd -
-```
+Since we have potentially multiple notary instances and multiple keys for each instance, Connaisseur somehow needs to know for which images it should use what kind of combination of instance and key. This is configured in the `policy` part of the `helm/values.yaml`, where different rules are defined, that need to be matched against a given image. Only one rule will be matched at the same time and this rule can specify which notary should be used, with the `notary` field. The value of this field has to match one of the notary `name` fields. Additionally a `key` field can be given in the rule, which will then specify which key within the chosen notary configuration should be used. This again has to match a value of a `name` field of one of the `rootKeys` entries. If no `notary` or `key` fields are given inside a rule, then the top most entry will be chosen respectively.
```

Comment:
```suggestion
Since we have potentially multiple notary instances and multiple keys for each instance, Connaisseur somehow needs to know for which images it should use what kind of combination of instance and key (otherwise, any key compromised in your supply chain would constitute a (potential) compromise of all your images). This is configured in the `policy` part of the `helm/values.yaml`, where different rules are defined, that need to be matched against a given image. Only one rule will be matched at the same time and this rule can specify which notary should be used, with the `notary` field. The value of this field has to match one of the notary `name` fields. Additionally a `key` field can be given in the rule, which will then specify which key within the chosen notary configuration should be used. This again has to match a value of a `name` field of one of the `rootKeys` entries. If no `notary` or `key` fields are given inside a rule, then the top most entry will be chosen respectively.
```

Reply:
resolved in 3b7d64cd6e6c5c09c2f7370d57ddb8a3b6c8869c